docs/src/index.md

```diff
@@ -5,11 +5,109 @@
 
 
 [ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl) helps you test [`ChainRulesCore.frule`](http://www.juliadiff.org/ChainRulesCore.jl/dev/api.html) and [`ChainRulesCore.rrule`](http://www.juliadiff.org/ChainRulesCore.jl/dev/api.html) methods, when adding rules for your functions in your own packages.
-
 For information about ChainRules, including how to write rules, refer to the general ChainRules Documentation:
 [![](https://img.shields.io/badge/docs-master-blue.svg)](https://JuliaDiff.github.io/ChainRulesCore.jl/dev)
 [![](https://img.shields.io/badge/docs-stable-blue.svg)](https://JuliaDiff.github.io/ChainRulesCore.jl/stable)
 
+## Canonical example
```

Comment:
should we make all of these doc tests or examples?
or make them all run and display examples?

Reply:
I know what doctests are. I don't know what "example" and "run and display examples" are, would you mind explaining?

I am a bit reluctant to add doctests because the printouts are pretty long and confusing, e.g.
```
julia> rrule_test(relu, 2.1, (-1.2, .23))
Test Summary:                      |
Don't thunk only non_zero argument | No tests
Test.DefaultTestSet("Don't thunk only non_zero argument", Any[], 0, false)
```

Reply:
I've added doctests